pandas/core/arrays/categorical.py

```diff
@@ -2441,19 +2441,24 @@ def replace(self, to_replace, value, inplace: bool = False):
         """
         inplace = validate_bool_kwarg(inplace, "inplace")
         cat = self if inplace else self.copy()
-        if to_replace in cat.categories:
-            if isna(value):
-                cat.remove_categories(to_replace, inplace=True)
-            else:
-                categories = cat.categories.tolist()
-                index = categories.index(to_replace)
-                if value in cat.categories:
-                    value_index = categories.index(value)
-                    cat._codes[cat._codes == index] = value_index
-                    cat.remove_categories(to_replace, inplace=True)
+        if is_list_like(to_replace):
```

Comment:
can you add some comments here on the procure for what you are doing.

Reply:
fixed, added comments around `replace_dict` generation